app/services/cocina_dro_access.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+# This transcodes the access from the item set rights form to Cocina DROAccess schema
+class CocinaDROAccess
+  extend Dry::Monads[:maybe]
+
+  # @param [String] the rights representation from the form (must be one of the keys in Dor::RightsMetadataDS::RIGHTS_TYPE_CODES, or 'default')
+  # @return [Maybe<Hash<Symbol,String>>] a hash representing a subset of the Access subschema of the Cocina model
+  def self.from_form_value(rights)
+    # Default only appears on the registration form, not the update form.
+    return None() if rights == 'default'
+
+    data = if rights == 'cdl-stanford-nd'
+             {
+               access: 'citation-only',
+               download: 'none',
+               controlledDigitalLending: true
+             }
+           elsif rights.end_with?('-nd') || %w[dark citation-only].include?(rights)
+             {
+               access: rights.delete_suffix('-nd'),
+               download: 'none',
+               controlledDigitalLending: false
+             }
+           elsif rights.start_with?('loc:')
+             {
+               access: 'location-based',
```

Comment:
This line looks uncovered by tests, according to CodeClimate, FYI.

Reply:
Just an FYI, to be clear, not a blocker.

Reply:
yeah, i went looking for where the other bits of this class might be tested so I could extend it, but it wasn't obvious to me